json/src/lib.rs

```diff
@@ -686,6 +686,59 @@ pub enum ImportMultiRequestScriptPubkey<'a> {
     Script(&'a Script),
 }
 
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub struct GetMempoolEntryResult {
+    /// Virtual transaction size as defined in BIP 141. This is different from actual serialized
+    /// size for witness transactions as witness data is discounted.
+    pub vsize: u64,
+    /// Transaction weight as defined in BIP 141. Added in Core v0.19.0.
+    pub weight: Option<u64>,
+    /// Local time transaction entered pool in seconds since 1 Jan 1970 GMT
+    pub time: u64,
+    /// Block height when transaction entered pool
+    pub height: u64,
+    /// Number of in-mempool descendant transactions (including this one)
+    #[serde(rename = "descendantcount")]
+    pub descendant_count: u64,
+    /// Virtual transaction size of in-mempool descendants (including this one)
+    #[serde(rename = "descendantsize")]
+    pub descendant_size: u64,
+    /// Number of in-mempool ancestor transactions (including this one)
+    #[serde(rename = "ancestorcount")]
+    pub ancestor_count: u64,
+    /// Virtual transaction size of in-mempool ancestors (including this one)
+    #[serde(rename = "ancestorsize")]
+    pub ancestor_size: u64,
+    /// Hash of serialized transaction, including witness data
+    pub wtxid: bitcoin::Txid,
+    /// Fee information
+    pub fees: GetMempoolEntryResultFees,
+    /// Unconfirmed transactions used as inputs for this transaction
+    pub depends: Vec<bitcoin::Txid>,
+    /// Unconfirmed transactions spending outputs from this transaction
+    #[serde(rename = "spentby")]
+    pub spent_by: Vec<bitcoin::Txid>,
```

Comment:
Does the RPC provide this field always? (I.e. when it's not spent, is it an empty list or not present?) If not, it might need `#[serde(default)]`.

Reply:
I verified that it does always provide this field, with an empty list if its unspent.